Gems/Atom/RPI/Code/Include/Atom/RPI.Edit/Common/JsonUtils.h

```diff
@@ -119,6 +123,62 @@ namespace AZ
                 return false;
             }
 
+            using DeserializedCpuData = AZStd::vector<RHI::CpuProfilingStatisticsSerializer::CpuProfilingStatisticsSerializerEntry>;
+            inline Outcome<DeserializedCpuData, AZStd::string> LoadSavedCpuProfilingStatistics(const AZStd::string& capturePath)
+            {
+                const auto* base = IO::FileIOBase::GetInstance();
+
+                char resolvedPath[IO::MaxPathLength];
+                base->ResolvePath(capturePath.c_str(), resolvedPath, IO::MaxPathLength);
+
+                // NOTE: this uses raw file pointers over the abstractions and utility functions provided by AZ::JsonSerializationUtils because
+                // saved profiling captures can be upwards of 400 MB. This necessitates a buffered approach to avoid allocating huge chunks of memory.
+                FILE* fp = nullptr;
+                azfopen(&fp, resolvedPath, "rb");
+                if (!fp)
+                {
+                    return Failure(AZStd::string::format("Could not fopen file %s, is the path correct?\n", resolvedPath));
+                }
+
+                constexpr AZStd::size_t BufSize = 65536;
+                char* buf = reinterpret_cast<char*>(azmalloc(BufSize));
```

Comment:
Note: this buffer is on the heap because it triggers VS analysis `C6262: Function uses '66792' bytes of stack: exceeds /analyze:stacksize '16384'. Consider moving data to heap.` when on the stack. I'm assuming there is no point in checking if `buf == nullptr` since much worse things will happen if a malloc call fails...

Reply:
That's unsurprising, that's a LOT of data to malloc!